### PR TITLE
[multitransport] Ignore unknown data

### PR DESCRIPTION
**Pull request by akarl10**
If the reserved filed is not 0 the request PDU seems to contain some extra data. Two bytes of 0 (probably a version field) followed by a JSON payload (not null terminated, until the end of the packet. There seems to be no dedicated length field)

This PDU was observed when connecting to a AVD over a gateway. The Json is probably metadata needed for a STUN/TURN setup of the UDP tunnels that are used by multitransport.

### CODE REVIEW

**Review by hardening — APPROVED** [review]
Looks ok for me (would need to have a look at what's going on in MS dlls regarding this)

**Comment by hardening** [comment]
@freerdp-bot can you test please

File: libfreerdp/core/multitransport.c [hunk 1]
```diff
@@ -63,13 +63,20 @@ state_run_t multitransport_recv_request(rdpMultitransport* multi, wStream* s)
...
 
 	Stream_Read_UINT32(s, requestId);      /* requestId (4 bytes) */
 	Stream_Read_UINT16(s, requestedProto); /* requestedProtocol (2 bytes) */
-	Stream_Seek(s, 2);                     /* reserved (2 bytes) */
+	Stream_Read_UINT16(s, reserved);       /* reserved (2 bytes) */
 	cookie = Stream_ConstPointer(s);
 	Stream_Seek(s, RDPUDP_COOKIE_LEN); /* securityCookie (16 bytes) */
+	if (reserved != 0)
+	{
+		WLog_WARN(TAG, "reserved is %d instead of 0, skipping %d bytes of unknown data", reserved,
```
Comment (akallabeth) [inline libfreerdp/core/multitransport.c:76]: illegal format string, `size_t` requires `PRIuz`

**Review by akallabeth — CHANGES_REQUESTED** [review]

File: libfreerdp/core/multitransport.c [hunk 2]
```diff
@@ -63,13 +63,20 @@ state_run_t multitransport_recv_request(rdpMultitransport* multi, wStream* s)
...
+		          Stream_GetRemainingLength(s));
+		Stream_Seek(s, Stream_GetRemainingLength(s));
```
Comment (akallabeth) [inline libfreerdp/core/multitransport.c:78]: `Stream_SafeSeek` might be appropriate, as we don´t know if the stream actually contains all that data

**Comment by akallabeth** [comment]
@akarl10 thank you, nice patch, but the comment for this pr might be better added in code so we know why it was added (or what is missing)

File: libfreerdp/core/multitransport.c [hunk 3]
```diff
@@ -63,13 +63,20 @@ state_run_t multitransport_recv_request(rdpMultitransport* multi, wStream* s)
...
 
 	Stream_Read_UINT32(s, requestId);      /* requestId (4 bytes) */
 	Stream_Read_UINT16(s, requestedProto); /* requestedProtocol (2 bytes) */
-	Stream_Seek(s, 2);                     /* reserved (2 bytes) */
+	Stream_Read_UINT16(s, reserved);       /* reserved (2 bytes) */
 	cookie = Stream_ConstPointer(s);
 	Stream_Seek(s, RDPUDP_COOKIE_LEN); /* securityCookie (16 bytes) */
+	if (reserved != 0)
+	{
+		WLog_WARN(TAG, "reserved is %d instead of 0, skipping %" PRIuz "bytes of unknown data",
```
Comment (akallabeth) [inline libfreerdp/core/multitransport.c:76]: same is true for  `UINT16` (`unsigned short`) which requires `PRIu16`

File: libfreerdp/core/multitransport.c (hunk 2 above)
Comment (hardening) [reply libfreerdp/core/multitransport.c:78]: I disagree except if I miss something `Stream_Seek(s, Stream_GetRemainingLength(s));` is always safe as Safe_Seek just checks `Stream_GetRemainingLength(s)`

Comment (akallabeth) [reply libfreerdp/core/multitransport.c:78]: @hardening true, forgot to add that with the log above `reserved` should contain the amount of remaining data?

File: libfreerdp/core/multitransport.c (hunk 3 above)
Comment (akarl10) [reply libfreerdp/core/multitransport.c]: ok

**Review by akallabeth — APPROVED** [review]
LGTM,
thank you for this contribution!

File: libfreerdp/core/multitransport.c (hunk 2 above)
Comment (akarl10) [reply libfreerdp/core/multitransport.c]: reserved does actually not contain the length but 1. So probably reserved is now a version field.

**Comment by akallabeth** [comment]
@freerdp-bot test again please

**Pull request merged**